jnigen/test/simple_package_test/lib/simple_package.dart

```diff
@@ -333,20 +377,348 @@ class Example1 extends jni.JObject {
   int whichExample() => _whichExample(reference).integer;
 }
 
-class _$Example1Type extends jni.JType<Example1> {
-  const _$Example1Type();
+class $Example1Type extends jni.JObjType<Example1> {
+  const $Example1Type();
 
   @override
   String get signature => r"Lcom/github/dart_lang/jnigen/pkg2/Example;";
+
+  @override
+  Example1 fromRef(jni.JObjectPtr ref) => Example1.fromRef(ref);
 }
 
 extension $Example1Array on jni.JArray<Example1> {
   Example1 operator [](int index) {
-    return Example1.fromRef(
-        elementAt(index, jni.JniCallType.objectType).object);
+    return ($type.elementType as $Example1Type)
+        .fromRef(elementAt(index, jni.JniCallType.objectType).object);
   }
 
   void operator []=(int index, Example1 value) {
     (this as jni.JArray<jni.JObject>)[index] = value;
   }
 }
+
+/// from: com.github.dart_lang.jnigen.generics.MyMap
+class MyMap<K extends jni.JObject, V extends jni.JObject> extends jni.JObject {
+  final $MyMapType<K, V> $type;
```

Comment:
Do we practically always use `$type`? If so it's fine to initialize in the constructor.

Otherwise, we could consider making it `late final` and initialize it the first time it is accessed.